service/src/test/java/bio/terra/landingzone/library/landingzones/management/ResourcesDeleteManagerTest.java

```diff
@@ -0,0 +1,171 @@
+  void delete_LZResourcesNotFound() throws LandingZoneRuleDeleteException {
+    mockResourceListing(Collections.emptyList(), Collections.emptyList());
+
+    resourcesDeleteManager.deleteLandingZoneResources(LANDING_ZONE_ID, RESOURCE_GROUP_NAME);
+
+    verify(deleteRulesVerifierMock, times(1)).checkIfRulesAllowDelete(any());
+    verify(genericResourcesManagerMock, never()).deleteById(any());
+  }
+
+  @Test
+  void delete_SomeLZResourcesExist() throws LandingZoneRuleDeleteException {
+    var genericResource = mock(GenericResource.class);
+    when(genericResource.tags())
+        .thenReturn(Map.of(LandingZoneTagKeys.LANDING_ZONE_ID.toString(), LANDING_ZONE_ID));
+    mockResourceListing(
+        Collections.emptyList() /*ignoring private endpoints*/,
+        Collections.singletonList(genericResource));
+
+    resourcesDeleteManager.deleteLandingZoneResources(LANDING_ZONE_ID, RESOURCE_GROUP_NAME);
+
+    verify(deleteRulesVerifierMock, times(1)).checkIfRulesAllowDelete(any());
+    verify(genericResourcesManagerMock, times(1)).deleteById(any());
+  }
+
+  @Test
+  void delete_LZResourceCouldNotBeDeletedBecauseItIsNotFound()
+      throws LandingZoneRuleDeleteException {
+    final String resourceId1 = "RESOURCE_ID_1";
+    final String resourceId2 = "RESOURCE_ID_2";
+    var genericResource1 =
+        mockGenericResource(
+            resourceId1, Map.of(LandingZoneTagKeys.LANDING_ZONE_ID.toString(), LANDING_ZONE_ID));
+    var genericResource2 =
+        mockGenericResource(
+            resourceId2, Map.of(LandingZoneTagKeys.LANDING_ZONE_ID.toString(), LANDING_ZONE_ID));
+    mockResourceListing(
+        Collections.emptyList() /*ignoring private endpoints*/,
+        Arrays.asList(genericResource1, genericResource2));
+
+    var managementException = mock(ManagementException.class);
+    var managementError = mock(ManagementError.class);
+    when(managementError.getCode()).thenReturn("ResourceNotFound");
+    when(managementException.getValue()).thenReturn(managementError);
+    doThrow(managementException).when(genericResourcesManagerMock).deleteById(resourceId1);
+
+    resourcesDeleteManager.deleteLandingZoneResources(LANDING_ZONE_ID, RESOURCE_GROUP_NAME);
+
+    verify(deleteRulesVerifierMock, times(1)).checkIfRulesAllowDelete(any());
+    verify(genericResourcesManagerMock, times(1)).deleteById(resourceId2);
+    verify(genericResourcesManagerMock, times(1)).deleteById(resourceId1);
+  }
+
+  @Test
+  void delete_CustomErrorDuringResourceDeletion() {
+    final String resourceId1 = "RESOURCE_ID_1";
+    final String customManagementExceptionCode = "CustomCode";
+    var genericResource1 =
+        mockGenericResource(
+            resourceId1, Map.of(LandingZoneTagKeys.LANDING_ZONE_ID.toString(), LANDING_ZONE_ID));
+
+    mockResourceListing(
+        Collections.emptyList() /*ignoring private endpoints*/,
+        Collections.singletonList(genericResource1));
+
+    var managementException = mock(ManagementException.class);
```

Comment:
Wonder if these should go in https://github.com/DataBiosphere/terra-landing-zone-service/blob/main/service/src/test/java/bio/terra/landingzone/stairway/flight/exception/translation/ExceptionFixtures.java#L6 for consistency?

Reply:
I refactored it for now a little bit to make them reusable, but I would keep them locally since those are specific. Don't want to introduce huge fixtures file.